numpyro/infer/util.py

```diff
@@ -59,7 +59,7 @@ def log_density(model, model_args, model_kwargs, params, skip_dist_transforms=Fa
     else:
         model = substitute(model, param_map=params)
     model_trace = trace(model).get_trace(*model_args, **model_kwargs)
-    log_joint = 0.
+    log_joint = jax.device_put(0.)
```

Comment:
@fehiepsi - maybe we should also add a check to `mask` so that it raises a ValueError if fn is not a callable to ensure that users don't trip up by not specifying this as a keyword:

```python
if fn is not None and not callable(fn):
    raise ValueError("Expected `fn` to be callable; instead found type(fn) = {}.".format(type(fn)))
```

Reply:
good point! I'll add it.